flintrock/scripts/download-hadoop.py

```diff
@@ -18,16 +18,27 @@
 
 if __name__ == '__main__':
     hadoop_version = sys.argv[1]
-    apache_mirror_url = "http://www.apache.org/dyn/closer.lua/hadoop/common/hadoop-{v}/hadoop-{v}.tar.gz?as_json".format(v=hadoop_version)
+    custom_mirror_url = sys.argv[2]
+
+    if custom_mirror_url:
+        mirror_url = custom_mirror_url.format(v=hadoop_version)
+    else:
+        mirror_url = "http://www.apache.org/dyn/closer.lua/hadoop/common/hadoop-{v}/hadoop-{v}.tar.gz?as_json"\
+            .format(v=hadoop_version)
 
     tries = 0
     while tries < 3:
-        mirror_info = json.loads(urlopen(apache_mirror_url).read().decode('utf-8'))
-        file_url = mirror_info['preferred'] + mirror_info['path_info']
-        file_name = os.path.basename(mirror_info['path_info'])
-        print('Downloading file at:', file_url)
+        if custom_mirror_url:
```

Comment:
Can we collapse these two branches somehow? I think parts like the `print()` and `urlretrieve()` should be written once in a common block after the `if`/`else` branch.

Same goes for the `if`/`else` branch just above--I'd prefer it if we could have just one branch in this whole file that does different stuff based on whether the URL is custom or not, and the rest of the code should be the same.